chain/network/src/peer/peer_actor.rs

```diff
@@ -161,10 +162,14 @@ impl PeerActor {
         force_encoding: Option<Encoding>,
         network_state: Arc<NetworkState>,
     ) -> anyhow::Result<actix::Addr<Self>> {
-        let connection_guard =
-            ConnectionGuard { event_sink: network_state.config.event_sink.clone() };
-
         let peer_addr = stream.peer_addr().context("stream.peer_addr()")?;
+        // WARNING: connection guard is reported AFTER peer_addr is resolved,
+        // so if resolving fails, Event::ConnectionClosed won't be emitted.
```

Comment:
I wonder if it would make sense to still construct the `connection_guard` above the `stream.peer_addr()` call, but make `ConnectionGuard::peer_addr` field optional, and similarly for `Event`?

Reply:
Whether it would make sense depends on whether some test would be able to reasonably interpret it. I do not expect any test to specifically check this system call to fail.